src/sqlfluff/rules/L009.py

```diff
@@ -91,15 +91,17 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             # We can't fail on a meta segment
             return None
         else:
-            # So this looks like the end of the file, but we
-            # need to check that each parent segment is also the last.
-            # We do this with reference to the templated file, because it's
-            # the best we can do given the information available.
-            file_len = len(context.segment.pos_marker.templated_file.templated_str)
-            pos = context.segment.pos_marker.templated_slice.stop
-            # Does the length of the file equal the end of the templated position?
-            if file_len != pos:
-                return None
+            # We know we are at a leaf of the tree but not necessarily at the end of the tree.
```

Comment:
Is this the same code that was added to L052? I wonder if this should be turned into a reusable function, e.g. on the rule context or base rule classes.

Reply:
Moved the logic to `BaseRule` and it looks much neater now 👍